File: istart-backend/models/Vote.js
const mongoose = require('mongoose');

const VoteSchema = new mongoose.Schema({
  user:        { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  startupIdea: { type: mongoose.Schema.Types.ObjectId, ref: 'StartupIdea', required: true },
  type:        { type: String, enum: ['upvote', 'downvote'], required: true },
}, { timestamps: true });

// One vote per user per idea
VoteSchema.index({ user: 1, startupIdea: 1 }, { unique: true });

module.exports = mongoose.model('Vote', VoteSchema);

File: istart_app/lib/models/vote.dart
// lib/models/vote.dart

class Vote {
  final String id;
  final String userId;
  final String ideaId;
  final String type; // 'upvote' | 'downvote'

  Vote({
    required this.id,
    required this.userId,
    required this.ideaId,
    required this.type,
  });

  factory Vote.fromJson(Map<String, dynamic> json) => Vote(
        id: json['_id'] ?? '',
        userId: json['user'] ?? '',
        ideaId: json['idea'] ?? '',
        type: json['type'] ?? '',
      );
}